scripts/utils/generate_strings.py

```diff
@@ -36,7 +37,31 @@ def construct_mapping(self, node, deep=False):
         return super().construct_mapping(node, deep)
 
 
-def parseTranslationStrings(yamlfile):
+def parseXLIFFTranslationStrings(xliff_file):
+    if not os.path.isfile(xliff_file):
+        exit(f"Unable to find {xliff_file}")
+
+    strings = {}
+
+    tree = etree.parse(xliff_file)
+    root = tree.getroot()
+
+    for node in root.findall('.//{urn:oasis:names:tc:xliff:document:1.2}trans-unit'):
+        # Remove any unexpected characters e.g. São Paulo -> SoPaulo
```

Comment:
```suggestion
        # Keep only ASCII alphabetic characters e.g. São Paulo -> SoPaulo
```

How does it work if we need to update a string with a new ID, e.g. for "cs" Czech Republic -> Czechia?

Reply:
The code is not aware of updates. As far as it is concerned, that would be a new string. IIUC that is also how Pontoon will treat a new string id right?

Reply:
Let's say you update the string and use a new ID in `extras.xliff`, e.g. `servers.cz` ⇾ `servers.cz2`, but then you'd need some mapping to still store the content of `servers.cz2` as `ServersCz`.

I suppose with this regex it would work out of the box, since numbers are removed, but it feels like an unexpected side effect than a planned behavior.

Reply:
Ok, it has taken me a while, but I think I see what you mean. 

If we need to change any of the root translations we can't just change the values is the `<source>` tags of the `extras.xliff` file, we actually need to create a new string id! 

This is a great call, but something we can handle when working on https://mozilla-hub.atlassian.net/browse/VPN-6350. The `extras.xliff` file is supposed to be automatically updated as the Mullvad server list is changed, we can add logic to create the mapping and so on in there. WDYT?

Reply:
That works for me. I just wanted to raise it now, so we don't paint ourselves into a corner.

Reply:
Perfect, I'll add a note about it in the ticket description so I don't forget either.